spec/support/logging_helper.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module LoggingHelper
+  def capture_log_tags
+    captured = []
+
+    original = SemanticLogger.method(:tagged)
+
+    allow(SemanticLogger).to receive(:tagged) do |tags, &blk|
+      captured << tags
+      blk ? blk.call : nil
+    end
+
+    yield
+
+    captured
+  ensure
+    allow(SemanticLogger).to receive(:tagged) do |*args, &blk|
+      original.call(*args, &blk)
+    end
+  end
```

Comment:
There's a `and_wrap_original` helper for mocking that would help tidy this up a little:

```suggestion
    captured = []

    allow(SemanticLogger).to receive(
      :tagged
    ).and_wrap_original do |original_method, **tags, &block|
      captured << tags
      original_method.call(**tags, &block)
    end

    yield

    captured
```